cfgov/core/utils.py

```diff
@@ -10,13 +10,26 @@
 from core.templatetags.svg_icon import svg_icon
 
 
-NON_GOV_LINKS = re.compile(
-    r"https?:\/\/(?:www\.)?(?![^\?]+\.gov)(?!(content\.)?localhost).*"
+LINK_PATTERN = re.compile(
+    r"^(?P<schema>https?)://(?P<domain>[^/:]+):?(?P<port>\d+)?(?P<path>/?.*)?$"
 )
+
 NON_CFPB_LINKS = re.compile(
     r"(https?:\/\/(?:www\.)?(?![^\?]*(cfpb|consumerfinance).gov)"
     r"(?!(content\.)?localhost).*)"
 )
+
+
+def should_interstitial(url: str) -> bool:
+    match = LINK_PATTERN.match(url)
+    if match.group("domain").endswith(".gov") and NON_CFPB_LINKS.match(url):
+        return False
+    return True
```

Comment:
It's a minor organizational thing, but could this function live below the compiled regexes in this file? It'd be nice to keep all of the regex definitions together.